src/actions/actions.ts

```diff
@@ -583,6 +589,21 @@ export class DeleteOperatorXVisual extends BaseOperator {
 }
 
 @RegisterAction
+export class UpperCaseOperatorXVisual extends BaseOperator {
+    public keys = ["U"];
+    public modes = [ModeName.Visual, ModeName.VisualLine];
+    
+    public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
+      [start, end] = adjustVimSelectionPosition(vimState, start, end);
```

Comment:
I actually don't understand why you need to transform the range here. Most of the logic in DeleteOperator is to ensure we get exactly the right newlines we need to delete, but we don't really care about capitalizing newlines. 

I think just running from start to end.getRight() should be enough?